lib/Parse/ParseType.cpp

```diff
@@ -417,6 +417,12 @@ ParserResult<TypeRepr> Parser::parseType(Diag<> MessageID,
   if (Tok.is(tok::arrow)) {
     // Handle type-function if we have an arrow.
     SourceLoc arrowLoc = consumeToken();
+    if(Tok.is(tok::kw_throws)) {
+     Diag<> DiagID = diag::throws_in_wrong_position;
+     diagnose(Tok.getLoc(), DiagID).fixItInsert(arrowLoc, "throws ")
+	     .fixItRemove(Tok.getLoc());
+     consumeToken();
```

Comment:
This is good, but I think we can probably recover as if the user wrote `throws` in the correct place by setting `throwsLoc`.